src/v/cluster/plugin_frontend.cc

```diff
@@ -223,14 +227,23 @@ ss::future<mutation_result> plugin_frontend::do_local_mutation(
 }
 
 errc plugin_frontend::validate_mutation(const transform_cmd& cmd) {
-    validator v(_topics, _table);
+    const auto& noproduce
+      = config::shard_local_cfg().kafka_noproduce_topics.value();
+    absl::flat_hash_set<model::topic> noproduce_topics;
+    for (const auto& topic_name : noproduce) {
+        noproduce_topics.emplace(topic_name);
+    }
+    validator v(_topics, _table, std::move(noproduce_topics));
```

Comment:
Unfortunately the audit log isn't in the noproduce topics list.  That's because this list is modifiable by the user and we want something stronger than this list: https://github.com/redpanda-data/redpanda/blob/54724bdc906525c29986b807d4e66e7635538872/src/v/kafka/server/handlers/produce.cc#L509-L511

So to not produce to the audit log you'd have to check that specifically.

Reply:
I see - I've just hardcoded a list of topic names here: _schemas, __audit, and __consumer_offsets

Reply:
Ah thanks, I've hardcoded a few topics now